canonical-web-and-design/ubuntu.com
# Cancel a subscription via the API.

## Done

- Add support for fetching the last purchase ids from the API.
- Add support for cancelling a contract via the API.
- Update the cancel component to cancel the contract.

## QA

- Visit [/advantage?test_backend=true](/advantage?test_backend=true) and log in.
- Click on a monthly contract (there is an issue at the moment where you can't select a monthly contract that you also have a yearly subscription for: https://github.com/canonical-web-and-design/commercial-squad/issues/253).
- Click 'Edit subscription'
- Click the 'You can cancel this subscription online or contact us.' link at the bottom.
- Type cancel and then click the red button.
- You subscription should get cancelled.


## Issue / Card

Fixes: https://github.com/canonical-web-and-design/commercial-squad/issues/116.
